src/Illuminate/Database/Migrations/Migrator.php

```diff
@@ -63,11 +75,13 @@ class Migrator
      */
     public function __construct(MigrationRepositoryInterface $repository,
                                 Resolver $resolver,
-                                Filesystem $files)
+                                Filesystem $files,
+                                Dispatcher $dispatcher = null)
     {
         $this->files = $files;
         $this->resolver = $resolver;
         $this->repository = $repository;
+        $this->events = $dispatcher;
```

Comment:
We like to arrange the dependencies based on length so please move this right above the repository.

```php

	        $this->files = $files;
	        $this->resolver = $resolver;
	        $this->events = $dispatcher;
	        $this->repository = $repository;
```

Reply:
While at it, maybe make it 

```
	        $this->files = $files;
	        $this->resolver = $resolver;
	        $this->repository = $repository;
	        $this->dispatcher = $dispatcher;
```

Reply:
No we specifically chose the phrasing "events" because it's the most used in the framework. Other usages should be renamed imo.

Reply:
As someone that type-hints A LOT the `Illuminate\Contracts\Bus\Dispatcher`, it sounds really weird to have a `$event->dispatch()` instead of $dispatcher->dispatch($event)`, but what do I know?

Reply:
> We like to arrange the dependencies based on length so please move this right above the repository.
> 
> ```
> 
> 	        $this->files = $files;
> 	        $this->resolver = $resolver;
> 	        $this->events = $dispatcher;
> 	        $this->repository = $repository;
> ```

Do you mean only for the assignment in the body, not the actual dependencies in the params?

The latter would be breaking.

Reply:
If its based on length, shouldn't it also be above `resolver`?

Reply:
Yeah you're right.

Reply:
Done